integration-tests/tests/src/tests/queries.ts

```diff
@@ -15,11 +15,884 @@
+      { type: "IntObject", value: [100] },
+    ],
+    tests: [
+      ["AssertLength", "IntObject", 1, "intCol == -1"],
+      ["AssertLength", "IntObject", 1, "intCol==0"],
+      ["AssertLength", "IntObject", 0, "1 == intCol"],
+      ["AssertLength", "IntObject", 2, "intCol != 0"],
+      ["AssertLength", "IntObject", 2, "intCol > -1"],
+      ["AssertLength", "IntObject", 3, "intCol >= -1"],
+      ["AssertLength", "IntObject", 2, "intCol < 100"],
+      ["AssertLength", "IntObject", 3, "intCol <= 100"],
+      ["AssertLength", "IntObject", 1, "intCol > 0x1F"],
+      ["AssertLength", "IntObject", 1, "intCol == $0", 100],
+      ["AssertLength", "IntObject", 2, "intCol >= 0 LIMIT(2)"],
+
+      ["AssertException", "IntObject", "Unsupported comparison between type", "intCol == false"],
+      ["AssertException", "IntObject", "Cannot convert", "intCol == 'not an int'"],
+      ["AssertException", "IntObject", "Unsupported comparison between type", "intCol == $0", "not an int"],
+      ["AssertException", "IntObject", "Unsupported comparison operator", "intCol BEGINSWITH 1"],
+      ["AssertException", "IntObject", "Unsupported comparison operator", "intCol CONTAINS 1"],
+      ["AssertException", "IntObject", "Unsupported comparison operator", "intCol ENDSWITH 1"],
+    ],
+  },
+
+  floatTests: {
+    schema: [{ name: "FloatObject", properties: [{ name: "floatCol", type: "float" }] }],
+    objects: [
+      { type: "FloatObject", value: [-1.001] },
+      { type: "FloatObject", value: [0.0] },
+      { type: "FloatObject", value: [100.2] },
+    ],
+    tests: [
+      ["AssertLength", "FloatObject", 1, "floatCol == -1.001"],
+      ["AssertLength", "FloatObject", 1, "floatCol = 0"],
+      ["AssertLength", "FloatObject", 0, "floatCol == 1"],
+      ["AssertLength", "FloatObject", 2, "floatCol != 0"],
+      ["AssertLength", "FloatObject", 2, "floatCol > -1.001"],
+      ["AssertLength", "FloatObject", 3, "floatCol >= -1.001"],
+      ["AssertLength", "FloatObject", 2, "floatCol < 100.2"],
+      ["AssertLength", "FloatObject", 3, "floatCol <= 100.2"],
+      ["AssertLength", "FloatObject", 1, "floatCol > 0x1F"],
+      ["AssertLength", "FloatObject", 1, "floatCol == $0", 100.2],
+      ["AssertLength", "FloatObject", 2, "floatCol >= 0.0"],
+
+      ["AssertException", "FloatObject", "Unsupported comparison between type", "floatCol == false"],
+      ["AssertException", "FloatObject", "Cannot convert", "floatCol == 'not a float'"],
+      ["AssertException", "FloatObject", "Unsupported comparison between type", "floatCol == $0", "not a float"],
+      ["AssertException", "FloatObject", "Unsupported comparison operator", "floatCol BEGINSWITH 1"],
+      ["AssertException", "FloatObject", "Unsupported comparison operator", "floatCol CONTAINS 1"],
+      ["AssertException", "FloatObject", "Unsupported comparison operator", "floatCol ENDSWITH 1"],
+
+      // ["Disabled", "AssertException", "FloatObject", "floatCol = 3.5e+38"],
```

Comment:
Not too sure what this is trying to test (there is no exception being thrown right now but I am not sure what expected behavior is these days considering this is from many years ago), should I remove this, is there meant to be support for "3e+12.." format? I can also add a regular test case for that. @takameyer @kraenhansen

Reply:
Interesting.  Perhaps it passes only in Node and not on React Native.  One would have to test.

Reply:
We had a conversation on Slack too:
> I'll lean towards having anything expressible as a JS `Number` should persist (as best as possible - with potential minor loss of precision) to the database file, without throwing.

Reply:
I think then I will add a proper test case for this and see what happens, ty all